jib-core/src/main/java/com/google/cloud/tools/jib/image/json/ManifestListGenerator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.image.json;
+
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
+import com.google.cloud.tools.jib.hash.Digests;
+import com.google.cloud.tools.jib.image.Image;
+import com.google.cloud.tools.jib.image.json.V22ManifestListTemplate.ManifestDescriptorTemplate;
+import java.io.IOException;
+import java.util.List;
+
+/** Generates a manifest list for {@link Image}s. */
+public class ManifestListGenerator {
+
+  private final List<Image> images;
+
+  public ManifestListGenerator(List<Image> images) {
+    this.images = images;
+  }
+
+  /**
+   * Generates a manifest list JSON for the given {@link Image}s.
+   *
+   * @param <T> child type of {@link BuildableManifestTemplate}
+   * @param manifestTemplateClass the JSON template to translate the image to
+   * @return a manifest list JSON
+   */
+  public <T extends BuildableManifestTemplate> ManifestTemplate getManifestListTemplate(
+      Class<T> manifestTemplateClass) {
+    try {
+      V22ManifestListTemplate manifestList = new V22ManifestListTemplate();
```

Comment:
I see we always create a Docker manifest list, which is another limitation of multi-platform image building. Note that there are 2 avenues for considering OCI vs Docker format.
- Base image: we take care of this automatically, and the format is transparent to the user.
- Target image(s): user-configurable with `<format>OCI` or `<format>Docker`.
That is, Jib will be broken when `<format>OCI` is specified (because we will build OCI manifests while a Docker manifest). We should add another item to fix. Please add this to your tracking list.

Reply:
Note that support OCI for target images is much more important: most registries can just convert OCI to Docker on-the-fly, so not accepting an OCI base image manifest isn't really a problem in practice. OTOH, specifying `<format>OCI` will outright breaking Jib image building (in the multi-platform case).

Reply:
Added